Board.java

```diff
@@ -51,9 +51,9 @@ public boolean checkForWin(char player) {
 
     public ArrayList availablePositions() {
         ArrayList<Integer> positions = new ArrayList<>();
-        for (int i=0; i < dimension; i++) {
+        for (int i=1; i < dimension; i++) {
             if (state.charAt(i) == '-') {
-                positions.add(i);
+                positions.add(i+1);
```

Comment:
so presumably i can be 0... so that means the UX says please choose a position from 0 to 8 ?  That isn't a nice UX.  Would be better to be 1 to 9 from UX perspective.  The Board though shouldn't be aware what the UX is doing, the position should be modified before the Board is told to place a mark in  a position.